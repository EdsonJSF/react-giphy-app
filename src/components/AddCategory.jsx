import { useState } from "react";

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Search Gif"
        value={inputValue}
        onChange={handleInputValue}
      />
    </form>
  );
};
