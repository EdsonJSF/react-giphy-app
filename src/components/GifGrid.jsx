import { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { getGifs } from "../../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  useEffect(() => {
    getGifs(category).then((gifsByCategory) => setGifs(gifsByCategory));
  }, [category]);

  return (
    <>
      <h3>{category}</h3>

      <ol>
        {gifs.map(({ id, title }) => (
          <li key={id}>{title}</li>
        ))}
      </ol>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
