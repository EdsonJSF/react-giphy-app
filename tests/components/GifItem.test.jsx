import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("test on <GiftItem />", () => {
  const title = "test";
  const url = "https://cdn-icons-png.flaticon.com/512/4838/4838856.png";

  test("should match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });
});
