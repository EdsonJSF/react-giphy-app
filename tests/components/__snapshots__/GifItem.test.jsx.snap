// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GiftItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://cdn-icons-png.flaticon.com/512/4838/4838856.png"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
